src/pages/ToDo.js

```diff
@@ -1,101 +1,120 @@
-import React from "react";
+import React, { createContext } from "react";
 import Form from "../components/todoClass/Form";
 import Notes from "../components/todoClass/Notes";
 import { v4 as uuidv4 } from "uuid";
-// import { Context } from "../components/context";
+// import { TodoProvider } from "../components/todoContext";
 
-// const contextValue = { tasks, value };
+export const ContextClass = createContext({});
 
 class Todo extends React.Component {
   state = {
     tasks: [],
     value: "",
   };
 
-  addTask = (task) => {
+  changeState = (e) => this.setState(e);
+
+  addTask = (task, tasks, changeState) => {
+    const taskEdit = this.editedTask(tasks);
     if (!task) return null;
-    this.setState(({ tasks }) => ({
+
+    if (taskEdit) {
+      return changeState({
+        tasks: tasks.map((task) => (task.id === taskEdit.id ? { ...task, edit: false } : task)),
+        value: "",
+      });
+    }
+
+    changeState(({ tasks }) => ({
       tasks: [
         ...tasks,
         {
           id: uuidv4(),
           title: task,
           done: false,
+          edit: false,
         },
       ],
       value: "",
     }));
   };
 
-  doneTask = (id) => {
-    this.setState(({ tasks }) => ({
+  doneTask = (id, changeState) => {
+    changeState(({ tasks }) => ({
       tasks: tasks.map((task) => {
         if (task.id === id) task.done = true;
         return task;
       }),
     }));
   };
 
-  deleteTask = (id) => {
-    this.setState(({ tasks }) => ({
+  deleteTask = (id, changeState) => {
+    changeState(({ tasks }) => ({
       tasks: tasks.filter((task) => task.id !== id),
     }));
   };
 
-  clearAll = () => {
-    this.setState(() => ({
+  clearAll = (changeState) => {
+    changeState(() => ({
       tasks: [],
     }));
   };
 
-  editTask = (id) => {
-    const { tasks } = this.state;
-    const selectedItem = tasks.find((task) => task.id === id);
+  editTask = (id, tasks, changeState) => {
+    const editTasks = tasks.map((task) => (task.id === id ? { ...task, edit: true } : task));
+    const selectedItem = (tasks = tasks.find((task) => task.id === id));
 
-    this.setState(({ tasks }) => ({
-      tasks: tasks.filter((task) => {
-        return task.id !== id;
-      }),
+    changeState(() => ({
       value: selectedItem.title,
+      tasks: editTasks,
     }));
   };
 
-  inputChange = (event) => {
-    this.setState({ value: event.target.value });
+  editedTask = (tasks = []) => {
+    let taskToEdit = tasks;
+    if (tasks.length > 0) {
+      taskToEdit = tasks.find((task) => task.edit);
+    }
+    // const taskToEdit = tasks.length > 0 ? tasks.filter((task) => task.edit) : [];
+    return taskToEdit && taskToEdit.length ? taskToEdit[0] : undefined;
   };
 
-  handleEnter = (event) => {
+  inputChange = (value, tasks, changeState) => {
```

Comment:
too much for onChange handler. I guess that you don't need such calculations handling onChange event